### PR TITLE
upgrade: upgrade to vulcanize cosmos v0.46 SMT 

### PR DESCRIPTION
**Pull request by gsk967**
Closes: #27 #15 

## Description

This pull request contains latest vulcanize [cosmos-sdk v0.46.0-smt-0.0.3-alpha](https://github.com/vulcanize/cosmos-sdk/releases/tag/v0.46.0-smt-0.0.3-alpha) SMT changes 

1. Remove auth handlers and added middlewares 
2. Add support SMT v0.46.0-smt-0.0.3-alpha 

______

For contributor use:

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added  relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

______

For admin use:

- [ ] Added appropriate labels to PR (ex. `WIP`, `R4R`, `docs`, etc)
- [ ] Reviewers assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

**Comment by gsk967** [comment]
@roysc  @i-norden  Please ignore `/x` nameservice,bond, and auction modules , those already reviewed 
Please review middleware code and remaining changes

File: x/evm/keeper/benchmark_test.go [hunk 1]
```diff
@@ -63,12 +65,12 @@ func DoBenchmark(b *testing.B, txBuilder TxBuilder) {
...
-		require.NoError(b, err)
+		// txData, err := types.UnpackTxData(msg.Data)
+		// require.NoError(b, err)
 
-		fees := sdk.Coins{sdk.NewCoin(suite.EvmDenom(), sdk.NewIntFromBigInt(txData.Fee()))}
-		err = authante.DeductFees(suite.app.BankKeeper, suite.ctx, suite.app.AccountKeeper.GetAccount(ctx, msg.GetFrom()), fees)
-		require.NoError(b, err)
+		// fees := sdk.Coins{sdk.NewCoin(suite.EvmDenom(), sdk.NewIntFromBigInt(txData.Fee()))}
+		// err = authante.DeductFees(suite.app.BankKeeper, suite.ctx, suite.app.AccountKeeper.GetAccount(ctx, msg.GetFrom()), fees)
+		// require.NoError(b, err)
```
Comment (roysc) [inline x/evm/keeper/benchmark_test.go:73]: this works with `authmiddleware`

File: x/evm/keeper/grpc_query_test.go [hunk 2]
```diff
@@ -467,7 +467,7 @@ func (suite *KeeperTestSuite) TestQueryValidatorAccount() {
 
 	for _, tc := range testCases {
 		suite.Run(fmt.Sprintf("Case %s", tc.msg), func() {
-			suite.SetupTest() // reset
+			// suite.SetupTest() // reset
```
Comment (roysc) [inline x/evm/keeper/grpc_query_test.go:470]: WIP?

File: x/evm/simulation/opereations.go [hunk 3]
```diff
@@ -0,0 +1,302 @@
+package simulation
```
Comment (roysc) [inline x/evm/simulation/opereations.go:1]: rename to operations.go

File: x/feemarket/migrations/v010/migrate_test.go [hunk 4]
```diff
@@ -22,12 +21,12 @@ import (
 func TestMigrateStore(t *testing.T) {
 	encCfg := encoding.MakeConfig(app.ModuleBasics)
 	feemarketKey := sdk.NewKVStoreKey(feemarkettypes.StoreKey)
-	tFeeMarketKey := sdk.NewTransientStoreKey(fmt.Sprintf("%s_test", feemarkettypes.StoreKey))
+	tFeeMarketKey := sdk.NewTransientStoreKey("margetkey_test")
```
Comment (roysc) [inline x/feemarket/migrations/v010/migrate_test.go:24]: ?

**Review by roysc — COMMENTED** [review]
Mostly just needs some cleanup. Also should have some TODO comments for things like the commented out IBC code in app, or anything else that's not fully refactored or removed.

File: x/feemarket/migrations/v010/migrate_test.go (hunk 4 above)
Comment (gsk967) [reply x/feemarket/migrations/v010/migrate_test.go:24]: `feemarkettypes.StoreKey` is already stored, It is storing the duplicate keys, thats why I added new StoreKey

File: app/app.go [hunk 5]
```diff
@@ -83,26 +90,32 @@ import (
...
+
+	// "github.com/cosmos/ibc-go/v3/modules/apps/transfer"
+	// ibctransferkeeper "github.com/cosmos/ibc-go/v3/modules/apps/transfer/keeper"
+	// ibctransfertypes "github.com/cosmos/ibc-go/v3/modules/apps/transfer/types"
+	// ibc "github.com/cosmos/ibc-go/v3/modules/core"
+	// ibcclient "github.com/cosmos/ibc-go/v3/modules/core/02-client"
+	// ibcclientclient "github.com/cosmos/ibc-go/v3/modules/core/02-client/client"
+	// porttypes "github.com/cosmos/ibc-go/v3/modules/core/05-port/types"
+	// ibchost "github.com/cosmos/ibc-go/v3/modules/core/24-host"
+	// ibckeeper "github.com/cosmos/ibc-go/v3/modules/core/keeper"
```
[the remark this answers is not in the document]
Comment (gsk967) [reply app/app.go]: This pr contains changes related to v0.46 => https://github.com/cosmos/ibc-go/pull/1205

**Pull request merged**